### PR TITLE
Use NetworkConnector directly to account for replaced hosts

### PR DESCRIPTION
**Pull request by nox**
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/16131)

### CODE REVIEW

**Comment by nox** [comment]
The failure is #14323.

File: components/url/lib.rs [hunk 1]
```diff
@@ -49,8 +49,8 @@ impl ServoUrl {
 
     // NOTE: These methods return options that are always true temporarily until
     // we special-case some urls to avoid going through rust-url.
```
Comment (jdm) [inline components/url/lib.rs]: This comment should go away if we're making this change. I don't understand it though; any comments, @emilio?

**Review by jdm — DISMISSED** [review]
This is a good change. I only have one comment about the changes in 22e4c9c.

File: components/url/lib.rs (hunk 1 above)
Comment (nox) [inline components/url/lib.rs]: @jdm This was from a time where we thought `ServoUrl` wouldn't always contain an `Url`, will remove it.

**Comment by nox** [comment]
@jdm r=you?

**Pull request merged**